middleware/http/wasm/basic/basic.go

```diff
@@ -137,3 +112,46 @@ func (m *Middleware) getNativeMetadata(metadata middleware.Metadata) (*middlewar
 
 	return &data, nil
 }
+
+type wapcRequestHandler struct {
+	mod  wapc.Module
+	pool *wapc.Pool
+}
+
+func (rh *wapcRequestHandler) requestHandler(h fasthttp.RequestHandler) fasthttp.RequestHandler {
+	return func(ctx *fasthttp.RequestCtx) {
+		instance, err := rh.pool.Get(1 * time.Second)
+		if err != nil {
+			ctx.Error("wasm pool busy", fasthttp.StatusInternalServerError)
+			return
+		}
+		defer instance.Close(ctx)
```

Comment:
The pool used here has a minor issue, especially maybe the close of the instance, it seems that the instances resued from the pool don't work as expected.

I updated the tests file to show and catch this error:

[patch.diff.zip](https://github.com/dapr/components-contrib/files/9266031/patch.diff.zip)

Reply:
thanks. I'll look into it and the problem wherever it is